source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesClusterOneTimeSetUp.cs

```diff
@@ -12,10 +13,27 @@ public async Task OneTimeSetUp()
     {
         var toolDownloader = new RequiredToolDownloader(KubernetesTestsGlobalContext.Instance.TemporaryDirectory, KubernetesTestsGlobalContext.Instance.Logger);
         var (kindExePath, helmExePath, kubeCtlPath) = await toolDownloader.DownloadRequiredTools(CancellationToken.None);
-        
+
         installer = new KubernetesClusterInstaller(KubernetesTestsGlobalContext.Instance.TemporaryDirectory, kindExePath, helmExePath, kubeCtlPath, KubernetesTestsGlobalContext.Instance.Logger);
         await installer.Install();
 
+        //if we are not running in TeamCity, then we need to find the latest local tag and use that if it exists 
+        if (!TeamCityDetection.IsRunningInTeamCity())
+        {
+            var imageLoader = new DockerImageLoader(KubernetesTestsGlobalContext.Instance.TemporaryDirectory, KubernetesTestsGlobalContext.Instance.Logger, kindExePath);
+            KubernetesTestsGlobalContext.Instance.TentacleImageAndTag = imageLoader.LoadMostRecentImageIntoKind(installer.ClusterName);
+        }
+        else
+        {
+            var tag = Environment.GetEnvironmentVariable("KubernetesAgentTests_ImageTag");
+            KubernetesTestsGlobalContext.Instance.TentacleImageAndTag = $"docker.packages.octopushq.com/octopusdeploy/kubernetes-tentacle:{tag}";
+        }
+
+        if (KubernetesTestsGlobalContext.Instance.TentacleImageAndTag is not null)
```

Comment:
Is there any case where this would actually be null? Since it gets set in the `else` statement above

Reply:
if there is no locally built image, it will be null coming out of the `DockerImageLoader`